src/Task/Git/BranchName.php

```diff
@@ -63,12 +64,14 @@ public function getConfigurableOptions()
     {
         $resolver = new OptionsResolver();
         $resolver->setDefaults([
-          'matchers' => [],
-          'additional_modifiers' => ''
+            'matchers' => [],
+            'additional_modifiers' => '',
+            'allow_symbolic_references' => true,
         ]);
 
         $resolver->addAllowedTypes('matchers', ['array']);
         $resolver->addAllowedTypes('additional_modifiers', ['string']);
+        $resolver->addAllowedTypes('allow_symbolic_references', ['boolean']);
```

Comment:
You are trying to search the branch name based on the symbolic reference 'HEAD'. Since in detached mode there is no symbolic reference HEAD, you retrieve the error. So I don't think the naming is correct here.
Maybe something like `allow_detached_head` is easier to understand?